src/CMakeLists.txt

```diff
@@ -113,6 +113,10 @@ if(NOT MSVC)
   add_compile_options($<$<COMPILE_LANGUAGE:CXX>:-Wno-ignored-qualifiers>)
 endif()
 
+if (CMAKE_CXX_COMPILER_ID STREQUAL Clang)
+  add_compile_options($<$<COMPILE_LANGUAGE:CXX>:-Wsign-compare>)
+endif()
```

Comment:
better off including references on what compiling options are covered now: we are using `-Wall` at this moment.
  - for clang, it does not cover `-Wsign-compare` which is covered by `-Wextra`, though. see https://clang.llvm.org/docs/DiagnosticsReference.html#wextra
  - for GCC, it covers `-Wsign-compare`, see https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wall